Lib/test/test_c_locale_coercion.py

```diff
@@ -100,11 +100,11 @@ def _set_locale_in_subprocess(locale_name):
 class EncodingDetails(_EncodingDetails):
     # XXX (ncoghlan): Using JSON for child state reporting may be less fragile
     CHILD_PROCESS_SCRIPT = ";".join([
-        "import sys, os",
-        "print(sys.getfilesystemencoding())",
-        "print(sys.stdin.encoding + ':' + sys.stdin.errors)",
-        "print(sys.stdout.encoding + ':' + sys.stdout.errors)",
-        "print(sys.stderr.encoding + ':' + sys.stderr.errors)",
+        "import sys, os, codecs",
+        "print(codecs.lookup(sys.getfilesystemencoding()).name)",
+        "print(codecs.lookup(sys.stdin.encoding).name + ':' + sys.stdin.errors)",
+        "print(codecs.lookup(sys.stdout.encoding).name + ':' + sys.stdout.errors)",
+        "print(codecs.lookup(sys.stderr.encoding).name + ':' + sys.stderr.errors)",
```

Comment:
Oh wait, I added _PyUnicode_InitEncodings() in Python 3.8: filesystem and stdio encodings are now always normalized. https://bugs.python.org/issue36775 :

```
commit 43fc3bb7cf0278735eb0010d7b3043775a120cb5
Author: Victor Stinner <vstinner@redhat.com>
Date:   Thu May 2 11:54:20 2019 -0400

    bpo-36775: Add _PyUnicode_InitEncodings() (GH-13057)
    
    Move get_codec_name() and initfsencoding() from pylifecycle.c to
    unicodeobject.c.
    
    Rename also "init" functions in pylifecycle.c.
```

So _handle_output_variations() can simply be removed from 3.8, without having to add codecs.lookup().name here.

This change makes sense to Python 3.7. Would you mind to first create a new PR for master, backport it to 3.8, and then retarget this PR to 3.7?

Reply:
Yes, that will be the reason why Python 3.8 no longer has this problem. I will look into it.

Is there a way how to retarget this to 3.7? When we tried it previously, Github tried to backport the whole master into 3.7... Or do I have to create new PR for that as well?

Reply:
Well, maybe creating a new PR for 3.7 is simpler :-)